binary_tree.c

```diff
@@ -128,6 +130,60 @@ static void tree_rotate_left(Tree* self, int value) {
     }
 }
 
+static void tree_rotate_right(Tree* self, int value) {
```

Comment:
why do you accept value? What if this value won't be found? I suggest you to accept `Tree*` and `Node*` which will be rotated to the right (it's assumed that it has right child). Also don't forget to check If the node is a root, in this case tree root should be updated